### PR TITLE
docs: clarify difference between jobinfos

### LINKED ISSUE #4656: Docs: clarify last_job_info vs detailed_job_info
### Describe the current issue

The `last_job_info` and `detailed_job_info` are different pieces of information stored on the node, acquired by different means.

The documentation of the functions is currently lacking in terms of how/when this information was acquired and how they differ.

```
    def get_detailed_job_info(self):
        """Return the detailed job info dictionary.
```

```
    def get_last_job_info(self):
        """Return the last information asked to the scheduler about the status of the job.
```

Note: we noticed that the status of the calculation reported in the `last_job_info` can differ from the status of the calculation reported in the `detailed_job_info` - in particular, the status reported in the `last_job_info` always seems to be `RUNNING`, while the `detailed_job_info` is acquired after the job finishes (?).

### PR DESCRIPTION
**Pull request by ltalirz**
fix #4656 

CalcJobNodes contain two different job infos, the detailed_job_info and
the last_job_info.

The distinction between the two was not obvious, and not documented.

### CODE REVIEW

**Comment by ltalirz** [comment]
@giovannipizzi 
It may actually be less confusing to update the `last_jobinfo` also at the very last poll (setting it to `None`), even if this means some "loss of information".
Anyhow, perhaps better not to change it now since people may rely on it...

**Review by giovannipizzi — DISMISSED** [review]
Thanks! I would personally keep the current behaviour (second-to-last) exactly because it's a different set of information (comes from `squeue` instead of `sacct`) and good to keep together with the detailed job info for debugging.

**Comment by giovannipizzi** [comment]
Do you know why the circleci docs fail?

**Comment by ltalirz** [comment]
thanks for the review - it seems the reference was not found; trying again

File: aiida/engine/processes/calcjobs/tasks.py [hunk 1]
```diff
@@ -179,11 +179,11 @@ async def do_update():
         if job_info is None:
             # If the job is computed or not found assume it's done
             node.set_scheduler_state(JobState.DONE)
-            job_done = True
         else:
             node.set_last_job_info(job_info)
             node.set_scheduler_state(job_info.job_state)
-            job_done = job_info.job_state == JobState.DONE
+
+        job_done = job_info.job_state == JobState.DONE
```
Comment (sphuber) [inline aiida/engine/processes/calcjobs/tasks.py:186]: This change will now raise if `job_info` is `None`, right? Don't see why this change is necessary

Comment (ltalirz) [reply aiida/engine/processes/calcjobs/tasks.py:186]: Oops, thanks for spotting!

I guess this should have been
```
job_done = node.get_scheduler_state() == JonState.DONE
```
but then it's a database access, so I will revert this

**Comment by chrisjsewell** [comment]
> Do you know why the circleci docs fail?

#4647

**Comment by ltalirz** [comment]
I think it should be fine to merge now

**Review by sphuber — APPROVED** [review]
Thanks @ltalirz
